backend/pkg/api/handle_schema_registry.go

```diff
@@ -0,0 +1,819 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/cloudhut/common/rest"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/redpanda-data/console/backend/pkg/console"
+	"github.com/redpanda-data/console/backend/pkg/schema"
+)
+
+func (api *API) handleSchemaRegistryNotConfigured() http.HandlerFunc {
+	type response struct {
+		IsConfigured bool `json:"isConfigured"`
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		rest.SendResponse(w, r, api.Logger, http.StatusOK, &response{IsConfigured: false})
+	}
+}
+
+func (api *API) handleGetSchemaRegistryMode() http.HandlerFunc {
+	if !api.Cfg.Kafka.Schema.Enabled {
+		return api.handleSchemaRegistryNotConfigured()
+	}
```

Comment:
Not relevant to the current state of things, but it might be worth calling out... I think that if we ever implement hot reloading for `Kafka.Schema` where users can enable `Kafka.Schema` and set other properties and we reload, these APIs will then still return the "not configured" responses? Not sure if it's worth future-proofing somehow perhaps, or just commenting somewhere?

Reply:
I'll leave this as is for now since we don't have any plans to introduce hot-reloading and I think this should immediately pop-up as an issue when testing this functionality.